packages/jitm/src/class-engine.php

```diff
@@ -0,0 +1,186 @@
+<?php
+/**
+ * Jetpack's JITM Engine class.
+ *
+ * @package automattic/jetpack-jitm
+ */
+
+namespace Automattic\Jetpack\JITMS;
+
+use Automattic\Jetpack\JITMS\Message;
+
+/**
+ * Class JITM\Engine
+ *
+ * Determines the rules of a JITM, which should display and when.
+ */
+class Engine {
+
+	/**
+	 * Is mobile browser
+	 *
+	 * @var bool $mobile_browser.
+	 */
+	private $mobile_browser;
+
+	/**
+	 * Returns the default rules
+	 *
+	 * @return array Default rules.
+	 */
+	public function default_rules() {
+		$rules = array_merge(
+			$this->preconnection_default_rules()
+		);
+
+		return apply_filters( 'jetpack_jitm_rules', $rules );
+	}
+
+	/**
+	 * Returns the pre-connection JITMs default rules
+	 *
+	 * @return array Pre-connection JITMs default rules.
+	 */
+	private function preconnection_default_rules() {
+
+		return array(
+			( new Message( 'jpsetup-posts', 'pre-connect' ) )
+				->user_is( 'administrator' )
```

Comment:
Since all of JITMs are meant to be shown to administrators, and since we want an administrator to connect the site, we can likely simplify and get rid of this.

Likewise, since all of the following filters are the same, I'd suggest just getting rid of those:  `->with_icon()`, `->priority( 100 )`, `->open_cta_in_same_window()`, `->is_dismissible( true )`, `->priority( 1000 )`, and `->is_hosted_with_partner( 'bluehost' )`.